cob_twist_controller/include/cob_twist_controller/interface_types/interface_type.h

```diff
@@ -0,0 +1,106 @@
+/*!
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_twist_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: June, 2015
+ *
+ * \brief
+ *   This header contains the class and method definitions of
+ *   all interface types (position/velocity).
+ *
+ ****************************************************************/
+#ifndef COB_CONTROL_COB_TWIST_CONTROLLER_INCLUDE_INTERFACE_TYPES_INTERFACE_TYPE_H_
+#define COB_CONTROL_COB_TWIST_CONTROLLER_INCLUDE_INTERFACE_TYPES_INTERFACE_TYPE_H_
+
+#include "cob_twist_controller/interface_types/interface_type_base.h"
+
+/* BEGIN InterfaceBuilder *****************************************************************************************/
+/// Class providing a static method to create damping method objects.
+class InterfaceBuilder
+{
+    public:
+        static InterfaceBase* create_interface(ros::NodeHandle& nh,
+                                               const TwistControllerParams &params);
+
+        InterfaceBuilder() {}
+        ~InterfaceBuilder() {}
+};
+/* END InterfaceBuilder *******************************************************************************************/
+
+
+
+
+/* BEGIN Interface_Velocity ****************************************************************************************/
+class Interface_Velocity : public InterfaceBase
+{
+    public:
+        Interface_Velocity(ros::NodeHandle& nh, const TwistControllerParams &params)
+        : InterfaceBase(nh,params)
+        {
+            pub_ = nh.advertise<std_msgs::Float64MultiArray>("joint_group_velocity_controller/command", 1);
+        }
+
+        ~Interface_Velocity() {}
+
+        virtual void process_result(const KDL::JntArray &q_dot_ik,
+                                    std::vector<double> &initial_position);
```

Comment:
does InterfaceVelocity need `initial_position`? or is this just to unify the interface?